pkg/controller/projectclaim/projectclaimadapter.go

```diff
@@ -350,12 +352,14 @@ func (c *ProjectClaimAdapter) IsRegionSupported() error {
 // EnsureRegionSupported modifies projectClaim.Status.State with result from IsRegionSupported.
 // If a region is not supported it returns an error and sets projectClaim.Status.State to ClaimStatusError.
 func (c *ProjectClaimAdapter) EnsureRegionSupported() (gcputil.OperationResult, error) {
-	if err := c.IsRegionSupported(); err != nil {
+	err := c.IsRegionSupported()
+	if err != nil {
 		c.projectClaim.Status.State = gcpv1alpha1.ClaimStatusError
-		c.StatusUpdate()
-		return gcputil.RequeueWithError(operrors.Wrap(err, ""))
 	}
-	return gcputil.ContinueProcessing()
+	if err == nil && c.projectClaim.Status.State == gcpv1alpha1.ClaimStatusError {
```

Comment:
why are we setting the state to Pending if there is a ClaimStatausError ?

Reply:
if it was error before, we set it back to pending if it is resolved